src/linear_algebra/bilinear_form.lean

```diff
@@ -1206,10 +1206,105 @@ begin
     ext, exact hm x }
 end
 
+/-- The restriction of a nondegenerate bilinear form `B` onto a submodule `W` is
+nondegenerate if `W ⊓ B.orthogonal W ≤ ⊥`. -/
+lemma nondegenerate_restrict_of_inf_orthogonal_le_bot
+  (B : bilin_form R₁ M₁) (hB : sym_bilin_form.is_sym B)
+  {W : submodule R₁ M₁} (hW : W ⊓ B.orthogonal W ≤ ⊥) :
```

Comment:
There's a choice to make here between `≤ ⊥` and `= ⊥`, and it looks like the linear algebra library uses `le_bot` slightly more. So let's keep it like this, unless you found that `= ⊥` was more helpful.

Reply:
Shouldn't this be stated in terms of `disjoint W (B.orthogonal W)`?

Reply:
I choose `≤ ⊥` since that is how `is_compl` is defined. Is there a strong reason to switch to `disjoint`?

Reply:
Yes, `disjoint` is defeq to `a ⊓ b ≤ ⊥`, but comes with a bunch of API that likely would make this lemma easier to use